jdaviz/configs/cubeviz/plugins/slice/slice.py

```diff
@@ -32,26 +35,30 @@ class Slice(PluginTemplateMixin):
     * :meth:`~jdaviz.core.template_mixin.PluginTemplateMixin.show`
     * :meth:`~jdaviz.core.template_mixin.PluginTemplateMixin.open_in_tray`
     * :meth:`~jdaviz.core.template_mixin.PluginTemplateMixin.close_in_tray`
-    * ``slice``
-      Current slice number.
     * ``value``
       Value (wavelength or frequency) of the current slice.  When setting this directly, it will
-      update automatically to the value corresponding to the nearest slice.
+      update automatically to the value corresponding to the nearest slice, if ``snap_to_slice`` is
+      enabled.
     * ``show_indicator``
       Whether to show indicator in spectral viewer when slice tool is inactive.
     * ``show_value``
       Whether to show slice value in label to right of indicator.
     """
+    _cube_viewer_cls = CubevizImageView
+    _cube_viewer_default_label = 'flux-viewer'
+    cube_viewer_exists = Bool(True).tag(sync=True)
+
+    allow_disable_snapping = Bool(False).tag(sync=True)  # noqa internal use to show and allow disabling snap-to-slice
+
     template_file = __file__, "slice.vue"
-    slice = Any(0).tag(sync=True)
-    value = Any(-1).tag(sync=True)
+    value = FloatHandleEmpty().tag(sync=True)
     value_label = Unicode("Wavelength").tag(sync=True)
-    value_unit = Any("").tag(sync=True)
+    value_unit = Unicode("").tag(sync=True)
+    value_editing = Bool(False).tag(sync=True)  # whether the value input is actively being edited
 
-    min_slice = Int(0).tag(sync=True)
-    max_slice = Int(100).tag(sync=True)
-    wait = Int(200).tag(sync=True)
+    slider_throttle = Int(200).tag(sync=True)  # milliseconds
```

Comment:
I noticed that clicking and dragging the slice indicator leaves it a fraction of a second behind my cursor, does that come from this throttle? Is there a particular reason 200 ms was chosen instead of something smaller?

Reply:
This throttle was already there (I just renamed it from `wait`).  This definitely will add a floor to the lag that is seen, but if the lag increases beyond this, we definitely should try to optimize the code so that it is more responsive.  Is this PR noticeably less responsive than main for you?

Reply:
No it's the same as `main`. Agreed on your point about coming back to this later and trying to optimize the code.